pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -367,32 +367,19 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
-	// failover mons running on host path to use persistent volumes if VolumeClaimTemplate is set and vice versa
+	// failover any mons present in the mon fail over list
 	for _, mon := range c.ClusterInfo.Monitors {
-		if c.HasMonPathChanged(mon.Name) {
-			logger.Infof("fail over mon %q due to change in mon path", mon.Name)
+		if c.monFailoverList.Has(mon.Name) {
```

Comment:
We need to make sure to sync the set access since it'll be across goroutines. I don't believe the `set` type is synchronized natively.

Reply:
We have a synchronization in place where both the mon `[start](https://github.com/rook/rook/blob/c80ea7b64424118c341de35611b68f4ac35a353d/pkg/operator/ceph/cluster/mon/mon.go#L175-L176)` (main thread) and the `[checkHealth](https://github.com/rook/rook/blob/c80ea7b64424118c341de35611b68f4ac35a353d/pkg/operator/ceph/cluster/mon/health.go#L155-L156)` run by acquiring the lock. 
So I think we are covered here. What do you think?